psi4/driver/procrouting/proc.py

```diff
@@ -1924,6 +1939,11 @@ def run_occ(name, **kwargs):
 
     occ_wfn = core.occ(ref_wfn)
 
+    # Shove variables into global space
+    if name in ['mp2', 'omp2', 'scs-omp2', 'scs(n)-omp2', 'scs-omp2-vdw', 'sos-omp2']:
```

Comment:
I don't quite see why we need a `if` in these?

Reply:
This and your previous comment are connected, so I'll answer together.

Indeed, I blithely though the `if ` was unnecessary, too! It turns out that if replacing (`P::e.globals` --> `wfn_->set_variabl`) and shoving instead of duplicating every set_variable to globals with one to wfn, one must either do the whole module (which I did for DFMP2 and DETCI) or prescribed bits (*mp2 for DFOCC, OCC, FNOCC). Otherwise what happens is that the SCF vars set ofnthe reference wfn are what get shoved back into globals, thereby replacing the desired `P::e.globals` set during the correlated calc.

Interestingly, there was trouble with `ci-multi` that might pop up again for other modules. That's a test with two rasci jobs with the wfn of the former starting the latter. Now that we're starting to set `wfn.energy()` on post-scf, the second job was taking the rasci energy as the HF energy. I changed how detci gets its ref energy, but we'll have to watch out.

So in answer to your previous comment, absolutely that's how it should be done, but it's unsafe for now.